### PR TITLE
Add Submission Confirmation Screen

### PR DESCRIPTION
**Pull request by rickreyhsig**
Add Submission Confirmation Screen to state file
https://www.pivotaltracker.com/story/show/186317924

Screen Shot 2023-11-09 at 2 39 15 PM

### CODE REVIEW

File: app/controllers/state_file/questions/submission_confirmation_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+      def edit; end
+
+      private
+
+      def form_class
+        NullForm
+      end
+    end
+  end
+end
```
Comment (rickreyhsig) [inline app/controllers/state_file/questions/submission_confirmation_controller.rb:13]: I'm a little unclear of what happens after this screen to be able to update the feature specs.

File: app/lib/navigation/state_file_az_question_navigation.rb [hunk 2]
```diff
@@ -21,6 +21,7 @@ class StateFileAzQuestionNavigation
       StateFile::Questions::AzCharitableContributionsController,
       StateFile::Questions::AzReviewController,
       StateFile::Questions::SubmitReturnController,
+      StateFile::Questions::SubmissionConfirmationController,
       StateFile::Questions::ConfirmationController
```
Comment (rickreyhsig) [inline app/lib/navigation/state_file_az_question_navigation.rb:25]:
```suggestion
```

File: app/lib/navigation/state_file_ny_question_navigation.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@ class StateFileNyQuestionNavigation
       StateFile::Questions::Ny214Controller,
       StateFile::Questions::UnemploymentController,
       StateFile::Questions::SubmitReturnController,
+      StateFile::Questions::SubmissionConfirmationController,
       StateFile::Questions::ConfirmationController
```
Comment (rickreyhsig) [inline app/lib/navigation/state_file_ny_question_navigation.rb:26]:
```suggestion
```

File: app/views/state_file/questions/submission_confirmation/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    <%= t(".help_friend_or_family") %>
+    <%= link_to 'Share the IRS DirectFile link', "https://www.google.com" %>
+  </p>
+
+  <p>
+    <%= link_to t(".register_to_vote"), "https://www.google.com" %>
+  </p>
+
+  <p>
+    <%= link_to t(".apply_to_state_benefits", state_name: current_intake.state_name), "https://www.google.com" %>
```
Comment (rickreyhsig) [inline app/views/state_file/questions/submission_confirmation/edit.html.erb:24]:
```suggestion
    # TODO - replace with real link
    <%= link_to t(".apply_to_state_benefits", state_name: current_intake.state_name), "" %>
```

File: app/views/state_file/questions/submission_confirmation/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    <%= t('.title', state_name: current_intake.state_name) %>
+  </h1>
+
+  <p>
+    <%= t(".email_text_update") %>
+  </p>
+
+  <p>
+    <%= t(".help_friend_or_family") %>
+    <%= link_to 'Share the IRS DirectFile link', "https://www.google.com" %>
```
Comment (rickreyhsig) [inline app/views/state_file/questions/submission_confirmation/edit.html.erb:16]:
```suggestion
    # TODO - replace with real link
    <%= link_to 'Share the IRS DirectFile link', "" %>
```

**Comment by bengolder** [comment]
- remove views/state_file/questions/confirmation/edit.html.erb
- remove ConfirmationController from the both navigation flows
- keep ConfirmationController
- add PDF download button ("Download your state return" )(above XML button)
- move page title into translation string ("The page with the submit button")
  - reuse `t('.title', state_name: current_intake.state_name)`
  - 'Share the IRS DirectFile link' also should go in translation string
- Instead of new `state_name` method, you can use `@state_name = States.name_for_key(params[:us_state])` in the view
- in feature spec, replace `expect(page).to have_text "You have successfully submitted your taxes"`
  with the title for the submission confirmation page

**Pull request merged**